i3lock.c

```diff
@@ -1004,105 +1004,148 @@ static void raise_loop(xcb_window_t window) {
+
+
+char *image_path = NULL;
+char *image_raw_format = NULL;
+int curs_choice = CURS_NONE;
+
+int set_opt(char o, char *optarg, int longoptind) {
+    switch (o) {
+        case 'v':
+            errx(EXIT_SUCCESS, "version " I3LOCK_VERSION " © 2010 Michael Stapelberg");
+        case 'n':
+            dont_fork = true;
+            break;
+        case 'b':
+            beep = true;
+            break;
+        case 'd':
+            fprintf(stderr, "DPMS support has been removed from i3lock. Please see the manpage i3lock(1).\n");
+            break;
+        case 'I': {
+                      fprintf(stderr, "Inactivity timeout only makes sense with DPMS, which was removed. Please see the manpage i3lock(1).\n");
+                      break;
+                  }
+        case 'c': {
+                      char *arg = optarg;
+
+                      /* Skip # if present */
+                      if (arg[0] == '#')
+                          arg++;
+
+                      if (strlen(arg) != 6 || sscanf(arg, "%06[0-9a-fA-F]", color) != 1)
+                          errx(EXIT_FAILURE, "color is invalid, it must be given in 3-byte hexadecimal format: rrggbb");
+
+                      break;
+                  }
+        case 'u':
+                  unlock_indicator = false;
+                  break;
+        case 'i':
+                  image_path = strdup(optarg);
+                  break;
+        case 't':
+                  tile = true;
+                  break;
+        case 'p':
+                  if (!strcmp(optarg, "win")) {
+                      curs_choice = CURS_WIN;
+                  } else if (!strcmp(optarg, "default")) {
+                      curs_choice = CURS_DEFAULT;
+                  } else {
+                      errx(EXIT_FAILURE, "i3lock: Invalid pointer type given. Expected one of \"win\" or \"default\".");
+                  }
+                  break;
+        case 'e':
+                  ignore_empty_password = true;
+                  break;
+        case 0:
+                  if (strcmp(longopts[longoptind].name, "debug") == 0)
+                      debug_mode = true;
+                  else if (strcmp(longopts[longoptind].name, "raw") == 0)
+                      image_raw_format = strdup(optarg);
+                  break;
+        case 'f':
+                  show_failed_attempts = true;
+                  break;
+        default:
+                  errx(EXIT_FAILURE, "Syntax: i3lock [-v] [-n] [-b] [-d] [-c color] [-u] [-p win|default]"
+                          " [-i image.png] [-t] [-e] [-I timeout] [-f]");
+    }
+
+    return 0;
+}
+
+int conf_file_rc() {
+
+    char buff[128] = {0};
+    char path[1024] = {0};
+    char *p = getenv("HOME");
+    int i = 0;
+
+    if(p == NULL)
+        return -ENOENT;
+
+    strcat(path, p);
+    strcat(path, "/.i3lockrc");
+    FILE *f = fopen(path, "r");
+
+    if(f == NULL)
+        return -ENOENT;
+
+    while(fgets(buff, sizeof buff, f) != NULL) {
```

Comment:
`while (fgets(buff, sizeof buff, f) != NULL)`

The `while` needs a space before the condition.